lib/auth/tun.go

```diff
@@ -553,6 +553,18 @@ func (s *AuthTunnel) passwordAuth(
 			}}
 		log.Infof("[AUTH] session authenticated prov. token: '%v'", conn.User())
 		return perms, nil
+	case AuthValidateTrustedCluster:
```

Comment:
the problem is this token gives permission to write any cert authority. this is a vulnerability. I think you should pass the token as a part of `ValidateRequest`, and instead of creating new role return `Nop` authenticated user with no privileges.  
in auth with roles you check specifically for the token scope though to modify just one CA.